node/status_node.go

```diff
@@ -337,10 +337,11 @@ func (n *StatusNode) Stop() error {
 func (n *StatusNode) stop() error {
 	if n.discoveryEnabled() {
```

Comment:
Maybe we can check here if discovery is also started? Because with the recent changes to the API, it's not obvious.